GCR/composite.py

```diff
@@ -338,12 +354,21 @@ def _obtain_native_data_dict(self, native_quantities_needed, native_quantity_get
                 continue
 
             # match column if needed:
+
             if cat.matching_by_column:
-                matching_idx, not_matched_mask, cat.sorter = _match(
-                    cat.cache[cat.matching_by_column],
-                    data[(self._main.identifier, cat.matching_column_in_main)],
-                    cat.sorter,
-                )
+
+                if cat.non_unique_match_selector:
+                    matching_idx, not_matched_mask, cat.sorter = _match(
+                        cat.cache[cat.matching_by_column],
+                        data[(self._main.identifier, cat.matching_column_in_main)],
+                        cat.sorter, cat.cache[cat.non_unique_match_selector], cat.select_descending
+                    )
+                else:
+                    matching_idx, not_matched_mask, cat.sorter = _match(
+                        cat.cache[cat.matching_by_column],
+                        data[(self._main.identifier, cat.matching_column_in_main)],
+                        cat.sorter,
+                    )
```

Comment:
```suggestion
                selector = None if cat.non_unique_match_selector is None else cat.cache[cat.non_unique_match_selector]
                matching_idx, not_matched_mask, cat.sorter = _match(
                    cat.cache[cat.matching_by_column],
                    data[(self._main.identifier, cat.matching_column_in_main)],
                    cat.sorter, selector, cat.select_descending
                )
```

Avoid repeating the function call in the code